manifests/server/instance/systemd.pp

```diff
@@ -0,0 +1,25 @@
+# @summary This define handles systemd drop-in files for the postgres main instance (default) or additional instances
+# @param service_name Overrides the default PostgreSQL service name.
+# @param systemd_drop_in_ensure sets the Systemd drop-in file to present or absent
+# @api private
+define postgresql::server::instance::systemd (
+  String[1] $service_name                       = $name,
+  Enum[present, absent] $systemd_drop_in_ensure = $postgresql::server::systemd_drop_in_ensure,
+  $port                                         = $postgresql::server::port,
+  $datadir                                      = $postgresql::server::datadir,
+  $extra_systemd_config                         = $postgresql::server::extra_systemd_config,
```

Comment:
Given this is a define, does it make sense to have any default at all (except perhaps for the ensure? You always pass them in for the specific instance anyway.

I'd also appreciate data types.

Reply:
I just implemented your suggestions. 
Regarding the data types, I tried to be consistent with #1397